beacon-chain/rpc/beacon_server.go

```diff
@@ -136,6 +148,109 @@ func (bs *BeaconServer) BlockTree(ctx context.Context, _ *ptypes.Empty) (*pb.Blo
+// state.latest_eth1_data is updated, and validator deposits up to this root can be processed.
+// The deposit root can be calculated by calling the get_deposit_root() function of
+// the deposit contract using the post-state of the block hash.
+func (bs *BeaconServer) eth1Data(ctx context.Context) (*pbp2p.Eth1Data, error) {
+	beaconState, err := bs.beaconDB.HeadState(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("could not fetch beacon state: %v", err)
+	}
+	// Fetch the current canonical chain height from the eth1.0 chain.
+	currentHeight := bs.powChainService.LatestBlockHeight()
+	eth1FollowDistance := int64(params.BeaconConfig().Eth1FollowDistance)
+
+	stateLatestEth1Hash := bytesutil.ToBytes32(beaconState.Eth1Data.DepositRoot)
+	// If latest ETH1 block hash is empty, send a default response
+	if stateLatestEth1Hash == [32]byte{} {
+		return bs.defaultDataResponse(ctx, currentHeight, eth1FollowDistance)
+	}
+	// Fetch the height of the block pointed to by the beacon state's latest_eth1_data.block_hash
+	// in the canonical eth1.0 chain.
+	_, stateLatestEth1Height, err := bs.powChainService.BlockExists(ctx, stateLatestEth1Hash)
+	if err != nil {
+		return nil, fmt.Errorf("could not verify block with hash exists in Eth1 chain: %#x: %v", stateLatestEth1Hash, err)
+	}
+	dataVotes := []*pbp2p.Eth1Data{}
+	voteCountMap := make(map[string]voteHierarchy)
+	bestVoteHeight := big.NewInt(0)
+	var mostVotes uint64
+	var bestVoteHash string
+	depContainers := bs.beaconDB.DepositsContainersTillBlock(ctx, nil)
+	for _, vote := range beaconState.Eth1DataVotes {
+		if ctx.Err() != nil {
+			return nil, ctx.Err()
+		}
+		eth1Hash := bytesutil.ToBytes32(vote.DepositRoot)
+		// Verify the block from the vote's block hash exists in the eth1.0 chain and fetch its height.
+		blockExists, blockHeight, err := bs.powChainService.BlockExists(ctx, eth1Hash)
+		if err != nil {
+			log.WithError(err).WithField("blockRoot", fmt.Sprintf("%#x", bytesutil.Trunc(eth1Hash[:]))).
+				Debug("Could not verify block with hash in ETH1 chain")
+			continue
+		}
+		if !blockExists {
+			continue
+		}
+		// Let dataVotes be the set of Eth1DataVote objects vote in state.eth1_data_votes where:
+		// vote.eth1_data.block_hash is the hash of an eth1.0 block that is:
+		//   (i) part of the canonical chain
+		//   (ii) >= ETH1_FOLLOW_DISTANCE blocks behind the head
+		//   (iii) newer than state.latest_eth1_data.block_data.
+		// vote.eth1_data.deposit_root is the deposit root of the eth1.0 deposit contract
+		// at the block defined by vote.eth1_data.block_hash.
+		isBehindFollowDistance := big.NewInt(0).Sub(currentHeight, big.NewInt(eth1FollowDistance)).Cmp(blockHeight) >= 0
+		isAheadStateLatestEth1Data := blockHeight.Cmp(stateLatestEth1Height) == 1
+		heightIdx := sort.Search(len(depContainers), func(i int) bool { return depContainers[i].Block.Cmp(currentHeight) >= 0 })
+		if heightIdx == 0 {
```

Comment:
Why continue?

Reply:
there is no deposit root to depositcontaner[-1] we can skip to the next vote